src/braket/circuits/free_parameter.py

```diff
@@ -0,0 +1,114 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import numbers
+from typing import Dict
+
+from sympy import Symbol
+
+
+class FreeParameter:
+    """
+    Class 'FreeParameter'
+    """
+
+    def __init__(self, name: str):
+        """
+        Initializes a new :class:'FreeParameter' object.
+        Free parameters can be used in parameterized circuits.
+
+        Args:
+            name: Name of the :class:'FreeParameter'. Can be a unicode value.
+
+        Examples:
+            >>> param1 = FreeParameter("theta")
+            >>> param1 = FreeParameter("\u03B8")
+        """
+        parameter_symbol = Symbol(name)
+        self._name = name
+        self._symbol = parameter_symbol
+        self._parameter_expression = parameter_symbol
+        self._parameter_values = []
+
+    @property
+    def name(self) -> str:
+        """
+        Returns:
+            str: Name for :class:'FreeParameter'.
+        """
+        return self._name
+
+    @property
+    def parameter_values(self) -> list:
+        """
+        Returns:
+            list: Parameter values for :class:'FreeParameter'.
+        """
+        return self._parameter_values
+
+    @property
+    def symbol(self) -> Symbol:
+        """
+        Returns:
+            Symbol: Symbol for :class:'FreeParameter'.
+        """
+        return self._symbol
+
+    def fix_values(self, param_values):
+        """
+        Sets the values for a :class:'FreeParameter'.
+
+        Args:
+            param_values: The values to be set for the object.
```

Comment:
How are circuits run if multiple values are fixed? Maybe change this to just a single value for now.

Reply:
Fixed to a singular value but the strategy for multiple value input will be implemented with the execution PR